src/analysis/bounded_variation.lean

```diff
@@ -634,6 +634,148 @@ lemma monotone_on.has_locally_bounded_variation_on {f : α → ℝ} {s : set α}
   has_locally_bounded_variation_on f s :=
 λ a b as bs, ((hf.evariation_on_le as bs).trans_lt ennreal.of_real_lt_top).ne
 
+
+
+namespace has_locally_bounded_variation_on
+
+variables {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
+
+/--
+The **signed** variation of `f` on the interval `Icc a b` intersected with the set `s`,
+assuming `f` has locally bounded variation on `s` (so that `to_real` doesn't squash `⊤`).
+-/
+@[nolint unused_arguments]
+noncomputable def variation_from_to (hf : has_locally_bounded_variation_on f s) (a b : α) : real :=
```

Comment:
Also, use `ℝ` instead of `real`.

Reply:
We should probably use `ereal` as the return type to make more sense without in the absence of `hf`. Just compose with [ereal.to_real](https://leanprover-community.github.io/mathlib_docs/data/real/ereal.html#ereal.to_real) when you need a real-valued function.

Reply:
My idea with both using `hf` and `real` was that I felt taking into account cases where the variation is not finite would be burdening ourselves for not much gain. If the variation is not finite, we lose additivity, and then it's probably easier to simply use `evariation_on`. Are we sure this level of generality is worth it? The only application I have in mind, beside the present refactor is arc-length parametrization, and in that case we only care about finite variation I'd say.

@sgouezel If I drop the `hf`, should it become **e**variation_**on**_from_to with `s` as an argument again?

@alreadydone so I'd lemmas like additivity talking about the composites?

Reply:
I'd stay with real, since it will be the main use case (and much more practical than ereal in which addition is borked). It would be named `variation_on_from_to` with arguments `f`, `s`, `a`, `b`, yes.

Reply:
In your three things going, I feel like 1. probably should not be changed for as long as possible.
For instance, I'm trying to figure out how to talk about "naturally parameterized" functions and such, and I think the easiest is 
```lean
def is_linearly_parameterized_on_by (f : ℝ → E) (s : set ℝ) (l : ℝ) :=
∀ x y, evariation_on f (s ∩ Icc x y) = ennreal.of_real (l * (y - x))

def is_naturally_parameterized_on (f : ℝ → E) (s : set ℝ) :=
is_linearly_parameterized_on_by f s 1

lemma is_linearly_parametrized_on_by_zero_iff (f : ℝ → E) (s : set ℝ) :
  is_linearly_parameterized_on_by f s 0 ↔ ∀ x y ∈ s, edist (f x) (f y) = 0 := sorry

example {f : ℝ → E} {s t : set ℝ} {φ : ℝ → ℝ}
  (φm : monotone_on φ s) (φst : s.maps_to φ t) (φst' : s.surj_on φ t) {l l' : ℝ}
  (hf : is_linearly_parameterized_on_by (f ∘ φ) s l) (hfφ : is_linearly_parameterized_on_by f t l')
  (x ∈ s) : s.eq_on φ (λ t, (l / l') * (t - x) + (φ x)) := sorry
```
and this would fit with the idea that we define path as functions from the reals that are eventually constant, and we prove things by restricting to appropriate sets of reals.